normandy/control/static/control/js/components/RecipeHistory.jsx

```diff
@@ -30,32 +30,64 @@ class RecipeHistory extends React.Component {
     const { revisionLog } = this.state;
     const { recipeId, recipe, dispatch } = this.props;
     return (
-      <div className="fluid-8">
+      <div className="fluid-8 recipe-history">
         <h3>Viewing revision log for: <b>{recipe ? recipe.name : ''}</b></h3>
         <ul>
-            {
-              this.state.revisionLog.map((revision, index) =>
-                <li key={revision.date_created} onClick={(e) => {
-                  let revisionInfo = {};
-                  if (index !== 0) {
-                    revisionInfo = {
-                      query: { revisionId: `${revision.id}` },
-                      state: { selectedRevision: revision.recipe }
-                    }
-                  }
-                  dispatch(push({
-                    pathname: `/control/recipe/${recipeId}/`,
-                    ...revisionInfo
-                  }))
-                }}><p className="revision-number">#{revision.recipe.revision_id} </p>
-                  <p><span className="label">Created On:</span>{ moment(revision.date_created).format('MMM Do YYYY - h:mmA') }</p>
-                </li>
-              )
-            }
+            {this.state.revisionLog.map((revision, index) =>
+              <HistoryItem
+                key={revision.id}
+                revision={revision}
+                recipe={recipe}
+                dispatch={dispatch}
+              />
+            )}
         </ul>
       </div>
     )
   }
 }
 
+class HistoryItem extends React.Component {
+  render() {
+    const { revision, recipe } = this.props;
+    const isCurrent = revision.recipe.revision_id === recipe.revision_id;
```

Comment:
Adding proptypes to the component will likely help track down the issue, and we should probably start doing that on new components.